File: cli/node/src/cli/commands/memory/get.ts
import type { JsonObject } from '../../../backend/json.js';
import type { Backend } from '../../../backend/types.js';
import { printError, timedStatus } from '../../../output/branding.js';
import { formatAgentEnvelope, formatSingleMemory } from '../../../output/format.js';
import { exit } from '../../../runtime/exit.js';
import { setCurrentCommand } from '../../../runtime/state.js';

/**
 * 获取指定记忆并按输出格式展示。
 * @param backend - 已装配的后端连接器。
 * @param memoryId - 目标记忆 ID。
 * @param opts - 本次操作选项。
 * @param opts.output - 终端输出格式。
 */
export async function cmdGet(
  backend: Backend,
  memoryId: string,
  opts: { output: string }
): Promise<void> {
  setCurrentCommand('get');
  let result: JsonObject;
  try {
    result = await timedStatus('Fetching memory...', async () => {
      return backend.get(memoryId);
    });
  } catch (e) {
    printError(e instanceof Error ? e.message : String(e));
    exit(1);
  }

  if (opts.output === 'agent') {
    formatAgentEnvelope({ command: 'get', data: result });
  } else {
    formatSingleMemory(result, opts.output);
  }
}

File: cli/python/src/memgo_cli/cli/commands/memory/get.py
"""记忆 get 用例。"""

from __future__ import annotations

import typer
from rich.console import Console

from memgo_cli.backend.types import Backend
from memgo_cli.output.branding import (
    print_error,
    timed_status,
)
from memgo_cli.output.format import (
    format_agent_envelope,
    format_single_memory,
)

console = Console()
err_console = Console(stderr=True)


def cmd_get(backend: Backend, memory_id: str, *, output: str) -> None:
    """读取并显示指定 ID 的记忆。"""
    from memgo_cli.runtime.state import is_agent_mode, set_current_command

    set_current_command("get")
    if is_agent_mode():
        output = "agent"
    with timed_status(err_console, "Fetching memory...") as _ts:
        try:
            result = backend.get(memory_id)
        except Exception as e:
            print_error(err_console, str(e), hint=None)
            raise typer.Exit(1) from None

    if output == "agent":
        format_agent_envelope(
            console, command="get", data=result, duration_ms=None, scope=None, count=None
        )
    else:
        format_single_memory(console, result, output)
